platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,157 @@
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.expiring.equals(existingAccount.expiring))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
+        existingAccount.lastObserved = observingAccount.lastObserved;
+      }
+    }
+  }
+
+  /**
+   * Remove the account from the observed account list.
+   *
+   * @param account The account being removed.
+   */
+  public void remove(String account) {
+    if (account != null) {
+      ObservingAccount targetAccount = allAccounts.get(account);
+      if (targetAccount != null) {
+        allAccounts.remove(account);
+      }
+    }
+  }
+
+  /**
+   * Gets the list of observed accounts.
+   *
+   * @return The list of observed accounts.
+   */
+  public List<ObservingAccount> getAccounts() {
+    return new ArrayList<>(allAccounts.values());
+  }
+
+  /**
+   * Check if the account is being observed. If the account is being observed, the lastObserved
+   * timestamp of the observing account will be updated.
+   *
+   * @param account The account to be checked.
+   * @return true if the account is being observed. false, otherwise.
+   */
+  public boolean observe(String account) {
+    ObservingAccount acct = allAccounts.get(account);
+    if (acct == null) return false;
+    acct.lastObserved = Instant.now();
+    return true;
+  }
+
+  /**
+   * Evict expired accounts
+   *
+   * @param maxAge evict all accounts that are older than maxAge
+   */
+  public void evict(Duration maxAge) {
+    for (ObservingAccount acct : getAccounts()) {
+      if (!acct.expiring) continue;
+
+      Duration age = Duration.between(Instant.now(), acct.lastObserved).abs();
+      if (age.compareTo(maxAge) > 0) {
+        allAccounts.remove(acct.account);
+        store.delete(acct.account);
+      }
+    }
+  }
+
+  @AllArgsConstructor
+  public static class ObservingAccount {
+    String account;
+    Instant lastObserved;
+    Boolean expiring;
+  }
+
```

Comment:
There are primarily two reasons.

`PaymentObservingAccount` is a ORM (ie: Hibernate) managed object. And `ObservingAccount` is stored and handled by the manager. If the ORM changes `account` field, this will break the integrity of the ConcurrentHashMap.

To avoid potential problems caused by ORM, I feel more comfortable to separate these two classes.

In addition, the PaymentObservingAccount table only stores expiring accounts. The manager has to handle all account.